### PR TITLE
Example: BarGroup with Tooltip

### PR DESCRIPTION
**Pull request by gksander**
Add an example of a custom tooltip with the BarGroup component.

https://github.com/FormidableLabs/victory-native-xl/assets/12721310/81ea079e-1f01-4500-aec6-4aaec3e560b5

### CODE REVIEW

**Comment by keithluchtel** [comment]
🔥

File: lib/src/cartesian/CartesianChart.tsx [hunk 1]
```diff
@@ -313,6 +316,12 @@ export function CartesianChart<
     ) as PointsArg;
   }, [_tData, yKeys]);
 
+  // On bounds change, emit
+  React.useEffect(() => {
+    onChartBoundsChange?.(chartBounds);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [chartBounds]);
```
Comment (carloskelly13) [inline lib/src/cartesian/CartesianChart.tsx:323]: My guess is it wants `onChartBoundsChange` to be part of this.

File: example/app/bar-with-tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                  <LinearGradient
+                    start={vec(0, 0)}
+                    end={vec(0, 500)}
+                    colors={["#fb9da0", "rgba(244,46,46,0.56)"]}
+                  />
+                </BarGroup.Bar>
+              </BarGroup>
+              {isActive && (
+                <>
+                  {/*<Blur clip={h} blur={1} />*/}
```
Comment (carloskelly13) [inline example/app/bar-with-tooltip.tsx:162]: Do we want to keep this?

File: lib/src/cartesian/CartesianChart.tsx (hunk 1 above)
Comment (gksander) [reply lib/src/cartesian/CartesianChart.tsx:323]: Yeah, but I want to allow an inline fn. I guess somewhere else we used a ref + effect to support this type of thing. Maybe we just roll our own sort of `useEvent`-like hook that can make this less ugly. I think I'll try to sneak this into this PR actually.

File: example/app/bar-with-tooltip.tsx (hunk 2 above)
Comment (gksander) [reply example/app/bar-with-tooltip.tsx:162]: ahh nope, good catch!

File: lib/src/hooks/useFunctionRef.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import * as React from "react";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const useFunctionRef = <Fn extends (...args: any[]) => any>(fn?: Fn) => {
```
Comment (carloskelly13) [inline lib/src/hooks/useFunctionRef.ts:4]: Could we change this to:
`Fn extends (...args: Parameters<Fn>) => ReturnType<Fn>`?

Comment (gksander) [reply lib/src/hooks/useFunctionRef.ts:4]: That looks like a recursive structure to me? This is how I typically do generic functions, and see them in the wild – ESLint is just really aggressive about these `any`s.

Comment (carloskelly13) [reply lib/src/hooks/useFunctionRef.ts:4]: Yeah, thinking of how we can do this without any. But maybe that's the play here.

**Pull request merged**